CodeGeneration/Sources/generate-swiftsyntax/templates/swiftsyntax/RawSyntaxNodesFile.swift

```diff
@@ -0,0 +1,273 @@
+                  guard var ptr = layout.baseAddress else { return }
+                  for elem in elements {
+                    ptr.initialize(to: elem.raw)
+                    ptr += 1
+                  }
+            }
+            self.init(raw: raw)
+          }
+          """
+        )
+
+        DeclSyntax(
+          """
+          public var elements: [Raw\(raw: node.collectionElementType.syntaxBaseName)] {
+              layoutView.children.map { Raw\(raw: node.collectionElementType.syntaxBaseName)(raw: $0!) }
+          }
+          """
+        )
+      }
+
+      if node.isBuildable || node.isMissing {
+        let params = FunctionParameterListSyntax {
+          for child in node.children {
+            FunctionParameterSyntax(
+              firstName: child.isUnexpectedNodes ? .wildcardToken(trailingTrivia: .space) : nil,
+              secondName: .identifier(child.swiftName),
+              colon: .colonToken(),
+              type: child.rawParameterType,
+              defaultArgument: child.isUnexpectedNodes ? child.defaultInitialization.map { InitializerClauseSyntax(value: $0) } : nil
+            )
+          }
+
+          FunctionParameterSyntax("arena: __shared SyntaxArena", for: .functionParameters)
+        }
+        try InitializerDeclSyntax("public init(\(params))") {
+          if !node.children.isEmpty {
+            for child in node.children {
+              if child.typeName == "TokenSyntax", case .token(let tokenChoices, _, _) = child.kind, !child.swiftName.contains("async"), !child.swiftName.contains("throw") {
+                let choices = tokenChoices.map { token -> (String, String) in
+                  switch token {
+                  case .keyword(let text):
+                    if text == "init" {
+                      return ("", ".keyword(.`init`")
+                    } else {
+                      return ("", ".keyword(.\(text))")
+                    }
+                  case .token(let tokenKind):
+                    return (".base", tokenKind)
+                  }
+                }
+                if child.isOptional {
+                  // FIXME: Assert disabled because normal CI doesn't have assertions enabled and debug CI seems to fail in a variety of places with this assertion
+                  // assert(${child.swift_name} == nil || ${' || '.join([f'{child.swift_name}?.tokenKind{base} == {choice}' for (base, choice) in choices])}, "Received \(String(describing: ${child.swift_name}?.tokenKind))")
+                } else {
+                  // FIXME: Assert disabled because normal CI doesn't have assertions enabled and debug CI seems to fail in a variety of places with this assertion
+                  // assert(${' || '.join([f'{child.swift_name}.tokenKind{base} == {choice}' for (base, choice) in choices])}, "Received \(${child.swift_name}.tokenKind)")
+                }
+              }
```

Comment:
Let’s just remove this entire section. I would like to re-land it in `RawSyntaxValidation.swift` anyway.